File: apps/web/lib/viewingContext.ts
export type LoginAudience = 'parent' | 'child';

const KEY_AUDIENCE = 'login_audience';
const KEY_READER_ID = 'selected_reader_id';
const KEY_READER_NAME = 'selected_reader_name';

export function setParentAudience() {
  localStorage.setItem(KEY_AUDIENCE, 'parent');
  localStorage.removeItem(KEY_READER_ID);
  localStorage.removeItem(KEY_READER_NAME);
}

export function setChildAudience(readerId: number, readerName: string) {
  localStorage.setItem(KEY_AUDIENCE, 'child');
  localStorage.setItem(KEY_READER_ID, String(readerId));
  localStorage.setItem(KEY_READER_NAME, readerName);
}

export function getAudience(): LoginAudience {
  if (typeof window === 'undefined') return 'parent';
  return localStorage.getItem(KEY_AUDIENCE) === 'child' ? 'child' : 'parent';
}

export function isChildAudience(): boolean {
  return getAudience() === 'child';
}

export function getSelectedReaderId(): number | null {
  if (typeof window === 'undefined') return null;
  const raw = localStorage.getItem(KEY_READER_ID);
  if (!raw) return null;
  const parsed = Number(raw);
  return Number.isFinite(parsed) ? parsed : null;
}

export function getSelectedReaderName(): string | null {
  if (typeof window === 'undefined') return null;
  return localStorage.getItem(KEY_READER_NAME);
}

export function clearViewingContext() {
  if (typeof window === 'undefined') return;
  localStorage.removeItem(KEY_AUDIENCE);
  localStorage.removeItem(KEY_READER_ID);
  localStorage.removeItem(KEY_READER_NAME);
}

File: vidhyavibeusermobileapp/lib/services/viewing_context.dart
import 'package:shared_preferences/shared_preferences.dart';

/// Parent vs child viewing mode — mirrors web viewingContext.ts
enum LoginAudience { parent, child }

class ViewingContext {
  ViewingContext._();

  static const _audienceKey = 'login_audience';
  static const _readerIdKey = 'selected_reader_id';
  static const _readerNameKey = 'selected_reader_name';

  static Future<void> setParentAudience() async {
    final prefs = await SharedPreferences.getInstance();
    await prefs.setString(_audienceKey, 'parent');
    await prefs.remove(_readerIdKey);
    await prefs.remove(_readerNameKey);
  }

  static Future<void> setChildAudience({
    required int readerId,
    required String readerName,
  }) async {
    final prefs = await SharedPreferences.getInstance();
    await prefs.setString(_audienceKey, 'child');
    await prefs.setString(_readerIdKey, readerId.toString());
    await prefs.setString(_readerNameKey, readerName);
  }

  static Future<LoginAudience> getAudience() async {
    final prefs = await SharedPreferences.getInstance();
    return prefs.getString(_audienceKey) == 'child'
        ? LoginAudience.child
        : LoginAudience.parent;
  }

  static Future<bool> isChildAudience() async {
    return await getAudience() == LoginAudience.child;
  }

  static Future<int?> getSelectedReaderId() async {
    final prefs = await SharedPreferences.getInstance();
    final raw = prefs.getString(_readerIdKey);
    if (raw == null) return null;
    return int.tryParse(raw);
  }

  static Future<String?> getSelectedReaderName() async {
    final prefs = await SharedPreferences.getInstance();
    return prefs.getString(_readerNameKey);
  }

  static Future<void> clear() async {
    final prefs = await SharedPreferences.getInstance();
    await prefs.remove(_audienceKey);
    await prefs.remove(_readerIdKey);
    await prefs.remove(_readerNameKey);
  }
}
